CONTRIBUTING.md

```diff
@@ -3,6 +3,62 @@
 We'd love to accept your patches and contributions to this project. There are
 just a few small guidelines you need to follow.
 
+## Contribution Guidelines
+
+To propose a new feature or a change that alters some existing user experience
+or creates a new user experience, follow these steps:
+
+### Step 1: Establish Context
+
+Search on KFP GitHub issues list to see if the same or similar proposal has been
+made in the past.  The historical context can help you draft a better
+proposal. Sometimes you will find a very similar proposal was already presented,
+discussed thoroughly, and that it is either awaiting contribution (in active
+development) or was rejected (often due to timing or conflicting scope with
+other plans). To avoid confusion and conflicts, where possible, please
+contribute to existing issues before creating new ones.
+
+### Step 2: Create Feature Request
+
+Create a new issue using the “Feature Request” template if no existing issue is
+found. Fill in answers to the template questions. To avoid delays, provide as
+much information as needed for initial review. Keep in mind that new features
+should comply with backward-compatibility and platform-portability requirements.
+
+### Step 3: Initial Team Triage
+
+Wait for a member from the Kubeflow Pipelines team (under
+orgs/kubeflow/teams/pipelines/ in
+[org.yaml](https://raw.githubusercontent.com/kubeflow/internal-acls/master/github-orgs/kubeflow/org.yaml))
+to comment on the issue. The team aims for triaging new issues on a weekly
+basis, but cannot at this time provide a guarantee on when your issue will be
+reviewed. The team will work with you to determine if your change is trivial and
+can proceed or whether it is nontrivial and needs a more detailed design
+document and review.
+
+### Step 4: Design Review
+
+If the team agreed with the overall proposal, you would be asked to write a
+design documentation, explaining why you want to make a change, what changes are
+you proposing, and how do you plan to implement it. The design review process
+would be required by default unless the team agreed that the change is too
+trivial. It is recommended that you use this Google doc template
+[[link](https://docs.google.com/document/d/1VrfuMo8ZeMmV75a-rUq9SO-E6KotBodAf-P0WZeFDZA/edit?usp=sharing&resourcekey=0-BklOgu8ivhdLCplZuPDZZg)]
```

Comment:
This link is not open to the public.

Reply:
This is shared with `kubeflow-discuss` google group as all other docs we have shared with the community.

Reply:
I see, should we make this doc fully public instead? People who land on this page tend to see this link first before joining the google group. (On the other hand, if you are not comfortable sharing it widely, you can also add a comment here to say this doc is shared with people who have joined Kubeflow community: https://www.kubeflow.org/docs/about/contributing/#joining-the-community)